schemas/json/layout/layout.schema.v1.json

```diff
@@ -0,0 +1,213 @@
+        }
+      },
+      "component": {
+        "type": "object",
+        "properties": {
+          "id": {
+            "type": "string",
+            "title": "id",
+            "description": "The component ID. Must be unique within a given layout."
+          },
+          "type": {
+            "type": "string",
+            "title": "type",
+            "description": "The component type.",
+            "enum": ["AddressComponent", "Paragraph", "Header", "Input", "TextArea", "Checkboxes", "RadioButtons", "Dropdown", "FileUpload", "Datepicker", "NavigationButtons", "Button", "Group"]
+          },
+          "required": {
+            "type": "boolean",
+            "title": "required",
+            "description": "Boolean value indicating if the component is required when filling in the form. Defaults to false."
+          },
+          "readOnly": {
+            "type": "boolean",
+            "title": "readOnly",
+            "description": "Boolean value indicating if the component should be presented as read only. Defaults to false."
+          },
+          "textResourceBindings": {
+            "type": "object",
+            "title": "textResourceBindings",
+            "description": "Text resource bindings for a component.",
+            "additionalProperties": {
+              "type": "string"
+            }
+          },
+          "dataModelBindings": {
+            "type": "object",
+            "title": "dataModelBindings",
+            "description": "Data model bindings for a component.",
+            "additionalProperties": {
+              "type": "string"
+            }
+          }
+        },
+        "required": ["id", "type"],
+        "allOf": [
+          { "if": {"properties": {"type": { "const": "Datepicker"}}}, "then": { "$ref": "#/definitions/datepickerComponent"}},
+          { "if": {"properties": {"type": { "const": "FileUpload"}}}, "then": { "$ref": "#/definitions/fileUploadComponent"}},
+          { "if": {"properties": {"type": { "const": "NavigationButtons"}}}, "then": { "$ref": "#/definitions/navigationButtonsComponent"}},
+          { "if": {"properties": {"type": { "const": "Group"}}}, "then": { "$ref": "#/definitions/groupComponent"}},
+          { "if": {"properties": {"type": { "const": "Dropdown"}}}, "then": { "$ref": "#/definitions/selectionComponents"}},
+          { "if": {"properties": {"type": { "const": "RadioButtons"}}}, "then": { "$ref": "#/definitions/selectionComponents"}},
+          { "if": {"properties": {"type": { "const": "Checkboxes"}}}, "then": { "$ref": "#/definitions/selectionComponents"}},
+          { "if": {"properties": {"type": { "const": "AddressComponent"}}}, "then": { "$ref": "#/definitions/addressComponent"}}
+        ]
+      },
+      "fileUploadComponent": {
+        "properties": {
+          "maxFileSizeInMB": {
+            "title": "maxFileSizeInMB",
+            "description": "Sets the maximum file size allowed in megabytes.",
+            "type": "integer"
```

Comment:
Add minimum value. Negative and zero should not be allowed. Applies to several of these properties.
https://json-schema.org/understanding-json-schema/reference/numeric.html#range